pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -159,7 +159,24 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	if lastTaskRun != nil {
 		// TODO: Make this safer
 		if len(lastTaskRun.Status.Conditions) > 0 {
+			buildRunStatus := buildRun.Status.Succeeded
 			taskRunStatus := lastTaskRun.Status.Conditions[0].Status
+
+			// check if we should delete the generated service account by checking the build run spec
+			// and that the build run status has not yet the completion state from the task run
+			if isGeneratedServiceAccountUsed(buildRun) &&
+				(buildRunStatus == "" || buildRunStatus == corev1.ConditionUnknown) &&
```

Comment:
we can rely purely on the taskrun status, so we should remove this line

Reply:
Done.